api/utils/iter/iter.go

```diff
@@ -0,0 +1,316 @@
+/*
```

Comment:
Interesting stuff here. Feels like a different language.

Since this is in the API module, what do you think about making it an internal package? Do we want people who import our API client to be able to use this stuff, or is it an implementation detail?

Note: I'm not sure we can even use an internal package since it is a separate module. I've never done a module-in-a-module outside of teleport before.

Reply:
We could put it in an `internal` subpackage to signal that the API is not to be relied upon.

Reply:
How about `api/unstable`.  I feel like that communicates the point very well.

Reply:
> Do we want people who import our API client to be able to use this stuff, or is it an implementation detail?

The helpers are mostly an implementation detail, but `GetInstances` will return `Iter[types.Instance]`.

Reply:
Alright, I've moved this to `api/unstable/iter` and added docs to `api/unstable` indicating that the sub-packages are present for dependency reasons only and don't necessarily adhere to the stability guarantees of the rest of `api`.

Reply:
Nevermind... `unstable` caused confusion.  Going w/ the original `internal` suggestion instead.